src/applicationManager.ts

```diff
@@ -513,9 +516,85 @@ export async function setResourceAccess({
       },
     );
     await errorHandler("Granting graph api permissions", assignRolesResult);
+  } else {
+    console.log("No graph api permissions to grant");
+  }
+}
+
+function _buildResourceAccess(roleIds: string[], accessType: string) {
+  // example: [{"id":"14dad69e-099b-42c9-810b-d002981feec1","type":"Scope"},
+  //           {"id":"e1fe6dd8-ba31-4d61-89e7-88639da4683d","type":"Role"}
+  //          ]
+  return roleIds.map((id) => ({ id, type: accessType }));
+}
+
+async function _buildRequiredResourceAccessPayload(currRequiredResourceAccessList: any[], graphAppId: string, appRoleIds: string[],
+                                                   scopeIds: string[]) {
+
+  /* todo: to remove current resource access and then build a new one based on access roles and access scopes. However we need to deal with
+           the eventual consistency of the graph API. */
+  let roleAccess = _buildResourceAccess(appRoleIds, "Role") ?? []
+  let scopeAccess = _buildResourceAccess(scopeIds, "Scope") ?? []
+  console.log("currRequiredResourceAccessObj", JSON.stringify(currRequiredResourceAccessList))
```

Comment:
is this still needed?

Reply:
I would keep it as it is useful in the log